### PR TITLE
feat:starting secondary server in a zone

### LINKED ISSUE #727: Secondary server listens in a Zone
**Is your feature request related to a problem? Please describe.**
A bad code path may cause the server to crash. Prevent the server to crash on a bad code path.

**Describe the solution you'd like**
Start the server in a zone

**Describe alternatives you've considered**
A simple try-catch, however, zoning is the recommended way and it has its own benefits.

**Additional context**
Please check this for an example:
https://dart.dev/articles/archive/zones#implementing-asynchronous-callbacks

runZoned(() {
  HttpServer.bind('0.0.0.0', port).then((server) {
    server.listen(staticFiles.serveRequest);
  });
},
onError: (e, stackTrace) => print('Oh noes! $e $stackTrace'));
Running the HTTP server in a zone enables the app to continue running despite uncaught (but non-fatal) errors in the server’s asynchronous code.

### PR DESCRIPTION
**Pull request by srieteja**
closes https://github.com/atsign-foundation/at_server/issues/727
**- What I did**
starting secondary server in a guarded zone
**- How I did it**
wrapped AtSecondaryServerImpl.start() in runZoneGaurded().
**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Review by VJag — APPROVED** [review]

File: at_secondary/at_secondary_server/lib/src/server/bootstrapper.dart [hunk 1]
```diff
@@ -48,7 +51,13 @@ class SecondaryServerBootStrapper {
       secondaryServerInstance.setExecutor(DefaultVerbExecutor());
       secondaryServerInstance
           .setVerbHandlerManager(DefaultVerbHandlerManager());
-      await secondaryServerInstance.start();
+
+      //starting secondary in a zone
+      runZonedGuarded(() async {
+        await secondaryServerInstance.start();
+      }, (error, stackTrace) {
+        logger.severe('Uncaught error: $error');
```
Comment (gkc) [inline at_secondary/at_secondary_server/lib/src/server/bootstrapper.dart]: @srieteja Can you do another PR please which also logs the stack trace?

Comment (srieteja) [reply at_secondary/at_secondary_server/lib/src/server/bootstrapper.dart]: @gkc I could add that additional log to this PR itself. Or should I do a new PR entirely?

Comment (gkc) [reply at_secondary/at_secondary_server/lib/src/server/bootstrapper.dart]: Oh yes, please add in this PR. Sorry I thought this PR had been closed already. Thanks @srieteja !

**Review by gkc — APPROVED** [review]

**Review by gkc — CHANGES_REQUESTED** [review]
The logging of the stack trace has been removed again ...

**Review by gkc — APPROVED** [review]
Oh I misread - I see you've combined it into one line.

**Comment by srieteja** [comment]
> The logging of the stack trace has been removed again ...

@gkc I have adjusted the log so that error and stacktrace are part of the same log. Rather than two different logs.
